schema.json

```diff
@@ -0,0 +1,118 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Thunderbird In-App Notifications",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "id": {
+        "type": "string",
+        "format": "uuid",
+        "description": "Unique ID set by the server"
+      },
+      "start_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "Timestamp after which Thunderbird will show the event after startup."
+      },
+      "end_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "Timestamp after which Thunderbird will never show the event, even if it has never been shown to the user."
+      },
+      "title": {
+        "type": "string",
+        "description": "Short sentence describing the event which will be displayed in the Thunderbird UI."
+      },
+      "description": {
+        "type": "string",
+        "description": "A short paragraph that can contain HTML and will be displayed in the Thunderbird UI."
+      },
+      "URL": {
+        "type": "string",
+        "format": "uri",
+        "description": "URL to load, if any."
+      },
+      "CTA": {
+        "type": "string",
+        "description": "Link text to show for the URL."
+      },
+      "severity": {
+        "enum": [1, 2, 3, 4, 5],
+        "description": "Severity level, where 1 is the most important/urgent and 5 is the least."
+      },
+      "type": {
+        "enum": ["donation", "message", "security", "blog"],
+        "description": "Category of notification."
+      },
+      "targeting": {
+        "type": "object",
+        "properties": {
+          "percent_chance": {
+            "type": "number",
+            "minimum": 0,
+            "maximum": 100
+          },
+          "exclude": {
+            "$ref": "#/definitions/profile"
+          },
+          "include": {
+            "$ref": "#/definitions/profile"
+          }
```

Comment:
Should these be arrays of profile in case we want to target specific combinations? I don't love the arrays of arrays structure this would lead to though.

Reply:
:thinking: making them arrays of `profile` is probably the best way to do combinations. But yeah, could be annoying to process on the client.